src/main/java/com/oya/kr/community/service/CommunityService.java

```diff
@@ -203,4 +229,54 @@ private User findByEmail(String email) {
 			.orElseThrow(() -> new ApplicationException(NOT_EXIST_USER));
 		return userMapperResponse.toDomain();
 	}
+
+	/**
+	 * 커뮤니티 게시글 스크랩
+	 *
+	 * @return String
+	 * @author 이상민
+	 * @since 2024.02.20
+	 */
+	public String saveCollection(String email, long communityId) {
+		User loginUser = findByEmail(email);
+		CollectionMapperRequest request = new CollectionMapperRequest(communityId, loginUser.getId());
+		int checkCollection = collectionMapper.check(request);
+		if(checkCollection == 0){
+			collectionMapper.save(request);
+			return "스크랩 완료되었습니다.";
+		}else if(checkCollection == 1){
+			boolean isDeleted = collectionMapper.isDeleted(request);
+			request.updateIsDeleted(isDeleted);
+			if(!isDeleted){
+				collectionMapper.delete(request);
+				return "스크랩 취소되었습니다.";
+			}else{
+				collectionMapper.delete(request);
+				return "스크랩 완료되었습니다.";
+			}
+		}else{
+			throw new ApplicationException(FAIL_COMMUNITY_COLLECTION);
+		}
+	}
+
+	/**
+	 * 카테고리 별 커뮤니티 게시글 분석 정보 조회
+	 *
+	 * @return String
+	 * @author 이상민
+	 * @since 2024.02.20
+	 */
+	public StatisticsResponse statistics() {
+		List<StatisticsResponseMapper> response = communityMapper.statistics();
+
+		Map<String, Integer> categoryCounts = response.stream()
+			.collect(Collectors.toMap(StatisticsResponseMapper::getCategoryCode, StatisticsResponseMapper::getCount));
+
+		List<StatisticsDetailResponse> list = new ArrayList<>();
+		for(Category category : Category.values()){
+			String code = category.getCode();
+			list.add(new StatisticsDetailResponse(code, categoryCounts.getOrDefault(code, 0)));
+		}
+		return new StatisticsResponse(list);
```

Comment:
이것도 stream 으로 바꿀 수 있을 것 같습니다!!!

```suggestion
		return new StatisticsResponse(
			Category.values().stream()
				.map(category -> new StatisticsDetailResponse(category.getCode(), categoryCounts.getOrDefault(code, 0)))
				.collect(toList());
```

인텔리제이가 없어 정확하지 않을 수 있지만... 참고 바랍니다 !! ㅎㅎ